src/Sentry/Platforms/Cocoa/Facades/SerializableNSObject.cs

```diff
@@ -0,0 +1,33 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+
+namespace Sentry.Cocoa.Facades;
+
+internal class SerializableNSObject : ISerializable
+{
+    private readonly NSObject _value;
+
+    public SerializableNSObject(NSObject value)
+    {
+        _value = value;
+    }
+
+    public Task SerializeAsync(Stream stream, IDiagnosticLogger? logger, CancellationToken cancellationToken = default)
+        => Task.Run(() => Serialize(stream, logger)); // TODO do we need a better implementation?
```

Comment:
If the code is synchronous we could call it synchronously and just return `Task.CompletedTask`?
This is called on the background worker class which is already a background task so no need to spawn another one with Task.Run

Reply:
thx, updated